components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/BasicAuthClientAuthenticator.java

```diff
@@ -200,8 +200,14 @@ protected boolean isBasicAuthorizationHeaderExists(HttpServletRequest request) {
         // authorizationHeader should be case-insensitive according to the
         // "The 'Basic' HTTP Authentication Scheme" spec (https://tools.ietf.org/html/rfc7617#page-3),
         // "Note that both scheme and parameter names are matched case-insensitively."
-        return StringUtils.isNotEmpty(authorizationHeader) &&
+        boolean isBasicAuthorizationHeaderExist = StringUtils.isNotEmpty(authorizationHeader) &&
                 authorizationHeader.toUpperCase().startsWith(BASIC_PREFIX.toUpperCase());
+        if (!isBasicAuthorizationHeaderExist) {
```

Comment:
Isn't this log is enough? As [this](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1734/files#diff-52bb68668533ce1e478cfb22a75d95f162ac4ba16166bc68a4d11498ab38abdeR219) condition also checked by this.

Reply:
This will check basic authorization header exists.  Other log I added to see whether authorization header comes in the request. Having both logs will give more idea